sources/packages/backend/libs/test-utils/src/factories/cas-supplier.factory.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  CASSupplier,
+  Student,
+  SupplierAddress,
+  SupplierStatus,
+  User,
+} from "@sims/sims-db";
+import {
+  createFakeUser,
+  E2EDataSources,
+  saveFakeStudent,
+} from "@sims/test-utils";
+
+/**
+ * Saves a fake CAS supplier.
+ * @param db manages the repositories to save the data.
+ * @param relations dependencies:
+ * - `student` student for the CAS supplier record.
+ * @param options optional params.
+ * - `supplierStatus` supplier status.
+ * - `isValid` valid flag.
+ * - `supplierAddress` supplier address.
+ * - `supplierNumber` supplier number.
+ * - `supplierProtected` supplier protected.
+ * @returns a saved fake CAS supplier.
+ */
+export async function saveFakeCASSupplier(
+  db: E2EDataSources,
+  relations?: {
+    student?: Student;
+  },
+  options?: {
+    supplierStatus?: SupplierStatus;
+    isValid?: boolean;
+    supplierAddress?: SupplierAddress;
+    supplierNumber?: string;
+    supplierProtected?: boolean;
```

Comment:
To avoid creating properties like in here we create the pattern of adding one Partial property of the target type.
```ts
options?: {
    initialValues: Partial<CASSupplier>;
}
```